apps/dashboard/app/(app)/settings/billing/stripe/page.tsx

```diff
@@ -45,43 +54,169 @@ export default async function StripeRedirect(props: Props) {
     typescript: true,
   });
 
-  // If they have a subscription already, we display the portal
-  if (ws.stripeCustomerId) {
-    const session = await stripe.billingPortal.sessions.create({
-      customer: ws.stripeCustomerId,
-      return_url: headers().get("referer") ?? "https://app.unkey.com",
-    });
+  const baseUrl = process.env.VERCEL_URL
+    ? process.env.VERCEL_TARGET_ENV === "production"
+      ? "https://app.unkey.com"
+      : `https://${process.env.VERCEL_URL}`
+    : "http://localhost:3000";
+  const returnUrl = headers().get("referer") ?? "https://app.unkey.com";
 
-    return redirect(session.url);
-  }
+  switch (props.searchParams.action) {
+    case "portal": {
+      if (!ws.stripeCustomerId) {
+        return (
+          <Empty>
+            <Empty.Title>No customer found</Empty.Title>
+            <Empty.Description>Your workspace</Empty.Description>
+            <Code>{ws.id}</Code>
+            <Empty.Description>
+              is not in Stripe yet. Please contact support@unkey.dev.
+            </Empty.Description>
+          </Empty>
```

Comment:
```ts
  <StripeError
            title="No customer found"
            description="Your workspace"
            code={ws.id}
          />
```

Can we make a component like this one? Looks like structure is pretty much the same.